site/src/examples/date-picker/ControlledOpen.tsx

```diff
@@ -0,0 +1,124 @@
+import type { OpenChangeReason } from "@floating-ui/react";
+import {
+  Button,
+  Divider,
+  FlexItem,
+  FlexLayout,
+  StackLayout,
+} from "@salt-ds/core";
+import type { DateFrameworkType } from "@salt-ds/date-adapters";
+import {
+  type DateInputSingleDetails,
+  DatePicker,
+  DatePickerActions,
+  DatePickerOverlay,
+  DatePickerSingleInput,
+  DatePickerSinglePanel,
+  DatePickerTrigger,
+  type SingleDateSelection,
+  useLocalization,
+} from "@salt-ds/lab";
+import {
+  type ReactElement,
+  type SyntheticEvent,
+  useCallback,
+  useRef,
+  useState,
+} from "react";
+
+export const ControlledOpen = (): ReactElement => {
+  const [open, setOpen] = useState(false);
+  const [selectedDate, setSelectedDate] = useState<
+    SingleDateSelection<DateFrameworkType> | null | undefined
+  >(null);
+  const { dateAdapter } = useLocalization();
+  const triggerRef = useRef<HTMLInputElement | null>(null);
```

Comment:
I don't think we need to explicitly add the `| null` type here
```suggestion
  const triggerRef = useRef<HTMLInputElement>(null);
```